### PR TITLE
InfluxQL specification update

### PR DESCRIPTION
**Pull request by gunnaraasen**
Bringing the InfluxQL spec up to date.  Currently a WIP.

### CODE REVIEW

**Comment by toddboom** [comment]
@gunnaraasen this is looking great so far

File: influxql/INFLUXQL.md [hunk 1]
```diff
@@ -103,15 +96,20 @@ VALUES       WHERE        WITH         WRITE
...
 
-InfluxQL supports decimal integer literals and float literals.  Hex, octal, etc. are not
-currently supported.
+InfluxQL supports decimal integer literals.  Hexadecimal and octal literals are not currently supported.
 
 ```
-int_lit             = decimal_lit .
-decimal_lit         = ( "1" .. "9" ) { decimal_digit } .
-float_lit           = decimals "." decimals .
+int_lit             = digit { digit } .
```
Comment (dgnorton) [inline influxql/INFLUXQL.md]: I don't think we should allow decimal literals to start with `0`.  I doubt we'll ever need to allow octal literals but old people like me might get confused if they see `0777` or similar.  :)

File: influxql/INFLUXQL.md [hunk 2]
```diff
@@ -613,14 +612,12 @@ binary_op        = "+" | "-" | "*" | "/" | "AND" | "OR" | "=" | "!=" | "<" |
 expr             = unary_expr { binary_op unary_expr } .
 
 unary_expr       = "(" expr ")" | var_ref | time_lit | string_lit |
-                   number_lit | bool_lit | duration_lit | regex_lit .
+                   float_lit | bool_lit | duration_lit | regex_lit .
```
Comment (dgnorton) [inline influxql/INFLUXQL.md]: also accepts `int_lit`

**Comment by toddboom** [comment]
@gunnaraasen it looks like your last commit actually got some of the merge conflict included. if you want to fix that up, we can go ahead and merge this.

**Comment by gunnaraasen** [comment]
@toddboom sorry, my branch got into a bad state. I think I managed to remove all the artifacts.

**Pull request merged**